include/flatcc/portable/pdiagnostic_push.h

```diff
@@ -29,12 +29,22 @@
 #if defined(_MSC_VER)
 #pragma warning( push )
 #define PDIAGNOSTIC_PUSHED_MSVC 1
-#elif defined(__clang__)
+#else
+#define PDIAGNOSTIC_PUSHED_MSVC 0
+#endif
+
+#if defined(__clang__)
 #pragma clang diagnostic push
 #define PDIAGNOSTIC_PUSHED_CLANG 1
-#elif ((__GNUC__ > 4) || (__GNUC__ == 4 && __GNUC_MINOR__ >= 6))
+#else
+#define PDIAGNOSTIC_PUSHED_CLANG 0
+#endif
+
+#if ((__GNUC__ > 4) || (__GNUC__ == 4 && __GNUC_MINOR__ >= 6))
```

Comment:
This doesn't test if __GNUC__ is defined. 
(If it is, it is reasonably to assume __GNUC_MINOR__ is also defined.)

Reply:
Also, it changes behaviour because it is no longer an alternative to the clang branch. __GNUC__ is defined for clang.

Reply:
i'm not sure what happened here, this is not the code i thought I committed.

Reply:
ah, i see... i thought i had change this already because it's so similar to logic in pdiagnostic.h, and upon re-reading it now to figure out what's going on I was so blind to the difference I thought I was going insane.